module/VuFind/src/VuFind/Search/Base/Results.php

```diff
@@ -554,6 +554,35 @@ public function getRawSuggestions()
         return $this->suggestions;
     }
 
+    /**
+     * Get extra data for the search.
+     *
+     * Extra data can be used to store local implementation-specific information.
+     * Contents must be serializable. It is recommended to make the array as small
+     * as possible.
+     *
+     * @return array
+     */
+    public function getExtraData(): array
+    {
+        // Not implemented in the base class
+        return [];
+    }
+
+    /**
+     * Set extra data for the search.
+     *
+     * @param array $data Extra data
+     *
+     * @return void
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function setExtraData(array $data): void
+    {
+        // Not implemented in the base class
```

Comment:
Should we throw an exception or log a warning if non-empty data is sent here?

Reply:
I'd rather not. It would cause vanilla VuFind to error with a search saved with Finna, and I'd rather let it silently ignore the data, even if it would mean it wouldn't represent the original search exactly.

Reply:
That's a good reason to avoid an exception, but what about a call to `error_log`? My concern is just avoiding a situation where parameters disappear inexplicably without any warning. If you think even that is too noisy, I can live without it, but just wanted to fully advocate for the possibility. :-)

Reply:
I suppose a simple error_log message is fine, though I'm not sure if that's necessary either.

Reply:
Log message added.

Reply:
Yeah, I really don't expect that this is ever likely to happen... but I just don't like having a setter that does absolutely nothing, because it seems like an opening for unexpected behavior. :-)